lib/ddtrace/configuration/option.rb

```diff
@@ -2,6 +2,7 @@
 module Datadog
   module Configuration
     # Represents an instance of an integration configuration option
+    # @public_api
     class Option
```

Comment:
Usual public question -- this seems to be the option system implementation; why public?

Reply:
Because `Option`'s public methods are used when uses create a configuration DSL for their custom instrumentations.

Reply:
Hmm... but do customers ever see these methods?

My thinking is that I'm looking at [`settings.rb`](https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/configuration/settings.rb) and even though those things get turned into `Options` instances -- we never actually see the `Options` class or any of its methods being called or referenced there. This was what led me to question this (and a few others) -- I understand that they get used, but it seems to me that they are the implementation details of mechanisms, and that customers, even advanced ones, are probably not going to need to know about them directly.